### PR TITLE
Split Mender binaries into 1.0.0b1 and git versions.

### PR DESCRIPTION
**Pull request by kacf**
Signed-off-by: Kristian Amlie <kristian.amlie@mender.io>

### CODE REVIEW

**Comment by kacf** [comment]
@pasinskim, @bboozzoo

**Comment by kacf** [comment]
Currently testing [here](https://ci.cfengine.com/job/yoctobuild-pr/335/).

**Comment by kacf** [comment]
Hmm, looks like this causes the version number to be 1.0.0b1 for *all* our builds, instead of a git revision for the dev builds. But I think we can fix that after the release.

**Comment by kacf** [comment]
> Hmm, looks like this causes the version number to be 1.0.0b1 for all our builds, instead of a git revision for the dev builds. But I think we can fix that after the release.

I have attempted to fix this in https://github.com/mendersoftware/mender-qa/pull/80, but it needs to be merged after this one.

**Comment by kacf** [comment]
Updated to fix acceptance test. Rerunning [here](https://ci.cfengine.com/job/yoctobuild-pr/337/).

File: meta-mender-core/classes/mender-install.bbclass [hunk 1]
```diff
@@ -63,6 +63,10 @@ MENDER_STORAGE_RESERVED_RAW_SPACE ??= "0"
 PREFERRED_VERSION_go-cross-arm ?= "1.7.%"
 PREFERRED_VERSION_go-native ?= "1.7.%"
 
+PREFERRED_VERSION_mender ?= "1.0%"
```
Comment (pasinskim) [inline meta-mender-core/classes/mender-install.bbclass]: Shouldn't we have here `PREFERRED_VERSION_mender ?= "1.0.%"`?

File: meta-mender-core/classes/mender-install.bbclass [hunk 2]
```diff
@@ -63,6 +63,10 @@ MENDER_STORAGE_RESERVED_RAW_SPACE ??= "0"
...
+PREFERRED_VERSION_mender-artifact ?= "1.0%"
+PREFERRED_VERSION_mender-artifact-native ?= "1.0%"
```
Comment (pasinskim) [inline meta-mender-core/classes/mender-install.bbclass]: Same as above. I am not sure if we need a dot after `1.0`.

File: meta-mender-core/recipes-mender/mender-artifact/mender-artifact_1.0.0b1.bb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require mender-artifact.inc
+
+SRC_URI = "git://github.com/mendersoftware/mender-artifact.git;protocol=https;branch=1.0.x"
+
+# Tag: 1.0.0b1
+SRCREV = "77326b288c70cd713e7ad15d2a084b6ee797e8ff"
```
Comment (pasinskim) [inline meta-mender-core/recipes-mender/mender-artifact/mender-artifact_1.0.0b1.bb]: Nice!

File: meta-mender-core/classes/mender-install.bbclass (hunk 1 above)
Comment (kacf) [inline meta-mender-core/classes/mender-install.bbclass]: You're right, we should have a dot. Will fix!

**Pull request merged**